src/scene_battle.cpp

```diff
@@ -93,10 +93,40 @@ void Scene_Battle::Start() {
 		return;
 	}
 
-	autobattle_algo = AutoBattle::CreateAlgorithm(Player::player_config.autobattle_algo.Get());
-	enemyai_algo = EnemyAi::CreateAlgorithm(Player::player_config.enemyai_algo.Get());
+	autobattle_algos.push_back(AutoBattle::CreateAlgorithm(AutoBattle::RpgRtCompat::name));
+	autobattle_algos.push_back(AutoBattle::CreateAlgorithm(AutoBattle::RpgRtImproved::name));
+	autobattle_algos.push_back(AutoBattle::CreateAlgorithm(AutoBattle::AttackOnly::name));
+	enemyai_algos.push_back(EnemyAi::CreateAlgorithm(EnemyAi::RpgRtCompat::name));
+	enemyai_algos.push_back(EnemyAi::CreateAlgorithm(EnemyAi::RpgRtImproved::name));
+
+	if (lcf::Data::system.easyrpg_default_actorai == -1 || (Player::debug_flag && Player::player_config.autobattle_algo.Get().size() > 0)) {
+		if (Player::player_config.autobattle_algo.Get().size() == 0) {
+			Player::player_config.autobattle_algo.Set(ToString(autobattle_algos[0]->GetName()));
+		}
+		for (auto& algo : autobattle_algos) {
+			if (algo->GetName() == Player::player_config.autobattle_algo.Get()) {
+				default_autobattle_algo = algo->GetId();
+				break;
+			}
+		}
+	} else {
+		default_autobattle_algo = lcf::Data::system.easyrpg_default_actorai;
+	}
+	if (lcf::Data::system.easyrpg_default_enemyai == -1 || (Player::debug_flag && Player::player_config.enemyai_algo.Get().size() > 0)) {
+		if (Player::player_config.enemyai_algo.Get().size() == 0) {
```

Comment:
@rueter37 Overall looks fine to me.
Can you replace all the ``size() > 0`` with ``!...empty()`` and ``size() == 0`` with ``empty()``?

Reply:
Fixed.